common/cmake/devices.cmake

```diff
@@ -272,12 +273,16 @@ function(DEFINE_DEVICE_TYPE)
   # If the UPDATE_TILES option is active, the architecture XML file will go through
   # an additional step and is updated to have include the `tiles` tags.
   #
+  # If ADD_PACK_PATTERNS option is active, the architecture XML file will go through
+  # an additional step to update some special `directs` with the correct pack pattern
+  # that is needed to successfully run the packing step.
+  #
   # DEFINE_DEVICE_TYPE defines a dummy target <arch>_<device_type>_arch that
   # will build the merged architecture file for the device type.
   #
   # If UPDATE_TIMINGS is set merged arch.xml file will be processed to inject
   # timing values using data from prjxray-db/$ARCH/timigs/*sdf files
-  set(options UPDATE_TIMINGS UPDATE_TILES)
+  set(options UPDATE_TIMINGS UPDATE_TILES ADD_PACK_PATTERNS)
```

Comment:
Why not use the already existing SCRIPTS mechanism?

Reply:
So, I have tried to use them, the problem was that if I add multiple scripts, cmake complains about the list that I pass to the `define_device_type` function

Reply:
So please fix that bug rather than add a new argument?

Reply:
FYI, you cannot pass a list of lists in CMake.  You can only pass the **name** of a list, and then copy that list during looping.  I can find an example if needed.

Reply:
Sure, I'll fix it than. Can you please provide the example?

Reply:
https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/make/project_xray.cmake#L483-L486

Reply:
Done